Makefile

```diff
@@ -17,6 +17,30 @@ WORKING_DIR     := $(shell pwd)
 EXAMPLES_DIR    := ${WORKING_DIR}/examples/yaml
 TESTPARALLELISM := 4
 
+OS := $(shell uname)
+EMPTY_TO_AVOID_SED := 
+
+prepare::
+	@if test -z "${NAME}"; then echo "NAME not set"; exit 1; fi
+	@if test -z "${REPOSITORY}"; then echo "REPOSITORY not set"; exit 1; fi
+	@if test -z "${ORG}"; then echo "ORG not set"; exit 1; fi
+	@if test ! -d "provider/cmd/pulumi-resource-x${EMPTY_TO_AVOID_SED}yz"; then "Project already prepared"; exit 1; fi
+
+	mv "provider/cmd/pulumi-resource-x${EMPTY_TO_AVOID_SED}yz" provider/cmd/pulumi-resource-${NAME}
+
+	if [[ "${OS}" != "Darwin" ]]; then \
+		find . \( -path './.git' -o -path './sdk' -o -path './examples' \) -prune -o -not -name 'go.sum' -type f -exec sed -i 's,github.com/pulumi/pulumi-[x]yz,${REPOSITORY},g' {} \; &> /dev/null; \
+		find . \( -path './.git' -o -path './sdk' -o -path './examples' \) -prune -o -not -name 'go.sum' -type f -exec sed -i 's/[xX]yz/${NAME}/g' {} \; &> /dev/null; \
+		find . \( -path './.git' -o -path './sdk' -o -path './examples' \) -prune -o -not -name 'go.sum' -type f -exec sed -i 's/[aA]bc/${ORG}/g' {} \; &> /dev/null; \
```

Comment:
Looks good but one small point about abc being replaced in the `provider.go` file: https://github.com/VenelinMartinov/my-provider/pull/2/files#diff-0e3b5cd2e6abf4633712dbd2d69307b7fbd75bc71844e34306afdcd9840dab64R84

I think you can use a different placeholder for the org name or exclude abcd from the replace pattern.

Reply:
I changed it so that it will just skip any lines containing 'SED_SKIP' which also cleans up the makefile.